EXPANDR-9733 - AWS Patching

capanw opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-dc.paloaltonetworks.com/browse/EXPANDR-9733

## Description
Files related to AWS Patching using SSM

## Must have
- [ ] Tests
- [ ] Documentation

johnnywilkes reviewed (changes requested):
made comments in Jira

johnnywilkes reviewed (changes requested):
@capanw , looking good, make sure @ShirleyDenkberg reviews, please

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+description: This playbook upgrades supported package on an AWS EC2 instance.
+id: 'AWS - EC2 Package Upgrade'
+inputSections:
+- description: Generic group for inputs
```
```suggestion
- description: Generic group for inputs.
```

capanw replied:
Updated!!

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+  - Assume Role
+  name: General (Inputs group)
+inputs:
+- description: ASM rule ID
```
```suggestion
- description: ASM rule ID.
```

capanw replied:
Updated!!

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+  required: true
+  value:
+    simple: ${alert.asmattacksurfaceruleid}
+- description: Instance ID of the EC2
```
```suggestion
- description: Instance ID of the EC2.
```

capanw replied:
Updated!!

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+  playbookInputQuery:
+  required: true
+  value: {}
+- description: AWS Region of the EC2 instance.
```
```suggestion
- description: AWS region of the EC2 instance.
```

capanw replied:
Updated!!

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+  playbookInputQuery:
+  required: true
+  value: {}
+- description: AWS Role to be assumed.
```
```suggestion
- description: AWS role to be assumed.
```

capanw replied:
Updated!!

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+  value: {}
+name: AWS - EC2 Package Upgrade
+outputSections:
+- description: Generic group for outputs
```
```suggestion
- description: Generic group for outputs.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+  - remediatedFlag
+outputs:
+- contextPath: remediatedFlag
+  description: Boolean value if Package is upgraded or not.
```
```suggestion
  description: Whether package is upgraded.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+    skipunavailable: false
+    task:
+      brand: ""
+      description: Determines if the AWS - Systems Manager integration instance is configured and Input values are defined.
```
```suggestion
      description: Determines if the AWS - Systems Manager integration instance is configured and input values are defined.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+    skipunavailable: false
+    task:
+      brand: ""
+      description: Checks if Package upgrade command run is initiated via AWS SSM.
```
```suggestion
      description: Checks if the package upgrade command run is initiated via AWS SSM.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Enrichment.yml:
```diff
@@ -753,50 +859,49 @@ view: |-
+      accessor: remoteip
+      root: alert
+  required: true
+  description: "AWS IP in alert"
```
```suggestion
  description: AWS IP address in the alert.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Enrichment.yml:
```diff
@@ -809,7 +914,26 @@ outputs:
+contentitemexportablefields:
+  contentitemfields: {}
+inputSections:
+- description: Generic group for inputs
```
```suggestion
- description: Generic group for inputs.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Enrichment.yml:
```diff
@@ -809,7 +914,26 @@ outputs:
+  - AWSAssumeRoleName
+  name: General (Inputs group)
+outputSections:
+- description: Generic group for outputs
```
```suggestion
- description: Generic group for outputs.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Enrichment_README.md:
```diff
@@ -10,30 +10,32 @@ This playbook does not use any sub-playbooks.
-| Indicator Query | Indicators matching the indicator query will be used as playbook input |  | Optional |
 | AwsIP | AWS IP in alert | alert.remoteip | Required |
 | AWSAssumeRoleName | If assuming roles for AWS, this is the name of the role to assume \(should be the same for all organizations\). |  | Optional |
+| Indicator Query | Indicators matching the indicator query will be used as playbook input |  | Optional |
```
```suggestion
| Indicator Query | Indicators matching the indicator query will be used as playbook input. |  | Optional |
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/README.md:
```diff
@@ -1,18 +1,19 @@
 The pack contains AWS playbooks that conduct enrichment and/or remediation and can use multiple other AWS content packs:
-- Enrichment: Give an IP address, see if there is a EC2 instance associated and if so pull information on the security group associated.
-- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved.
+- Enrichment: Give an IP address, see if there is a EC2 instance associated and if so pull information on the security group associated. If an SSM agent is installed on the instance, then pull information from SSM agent.
```
```suggestion
- Enrichment: Give an IP address, see if there is an associated EC2 instance and if so pull information on the associated security group. If an SSM agent is installed on the instance, pull information from the SSM agent.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/README.md:
```diff
@@ -1,18 +1,19 @@
-- Enrichment: Give an IP address, see if there is a EC2 instance associated and if so pull information on the security group associated.
-- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved.
+- Enrichment: Give an IP address, see if there is a EC2 instance associated and if so pull information on the security group associated. If an SSM agent is installed on the instance, then pull information from SSM agent.
+- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved. If there is a matching ASM rule ID that is supported, then upgrade the package associated to latest available version.
```
```suggestion
- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved. If there is a matching ASM rule ID that is supported, upgrade the associated package to the latest available version.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/README.md:
```diff
@@ -1,18 +1,19 @@
 - Unclaimed S3 Bucket Remediation: The playbook will create the unclaimed S3 bucket.
 
-There are multiple AWS content packs for multiple AWS products (EC2, IAM, Route53, S3, etc).  The intent was that users can install and use only the packs they need.  However, if an AWS playbook uses multiple pack integrations (such as EC2, S3 and IAM), the integrations can't reside in one of the current packs because they include content from multiple pack integrations.  This pack was created as ...
+There are multiple AWS content packs for multiple AWS products (EC2, IAM, Route53, S3, SSM etc).  The intent was that users can install and use only the packs they need.  However, if an AWS playbook uses multiple pack integrations (such as EC2, S3, SSM and IAM), the integrations can't reside in one of the current packs because they include content from multiple pack integrations.  This pack was c ...
```
```suggestion
There are multiple AWS content packs for multiple AWS products (EC2, IAM, Route53, S3, SSM, etc.).  The intent was that users can install and use only the packs they need. However, if an AWS playbook uses multiple pack integrations (such as EC2, S3, SSM, and IAM), the integrations can't reside in one of the current packs because they include content from multiple pack integrations. This pack was created as a place to put AWS playbooks that use AWS integrations from multiple packs with a focus on enrichment and remediation.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/README.md:
```diff
@@ -24,6 +25,11 @@ AWS - Enrichment playbook reports EC2 information given an IP address of an EC2
 ![AWS - Enrichment](https://raw.githubusercontent.com/demisto/content/master/Packs/AWS-Enrichment-Remediation/doc_files/AWS_-_Enrichment.png)
 
+#### AWS - EC2 Package Upgrade
+AWS - EC2 Package Upgrade playbook upgrades supported package on an AWS EC2 instance.
```
```suggestion
AWS - EC2 Package Upgrade playbook upgrades supported packages on an AWS EC2 instance.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_18.md:
```diff
@@ -0,0 +1,16 @@
+
+##### AWS - Enrichment
+
+- Updated the playbook to enrich using AWS SSM.
```
```suggestion
Updated the playbook to enrich using AWS SSM.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_18.md:
```diff
@@ -0,0 +1,16 @@
+
+##### New: AWS - EC2 Package Upgrade
+
+- New: This playbook upgrades supported package on an AWS EC2 instance.
```
```suggestion
New: This playbook upgrades supported package on an AWS EC2 instance.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_18.md:
```diff
@@ -0,0 +1,16 @@
+
+##### New: AWSEC2PackageUpgrade
+
+- New: This is an AWS script that upgrades a package on the AWS EC2 instance.
```
```suggestion
New: This is an AWS script that upgrades a package on the AWS EC2 instance.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.yml:
```diff
@@ -0,0 +1,53 @@
+name: AWSEC2PackageUpgrade
+outputs:
+- contextPath: awsec2packageupgrade
+  description: This is the command Id of the command initiated by AWS SSM command.
```
```suggestion
  description: The command ID of the command initiated by the AWS SSM command.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/README.md:
```diff
@@ -0,0 +1,44 @@
+---
+This script is used in the following playbooks and scripts.
+
+* AWS - EC2 Package Upgrade
```
```suggestion
AWS - EC2 Package Upgrade
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/README.md:
```diff
@@ -0,0 +1,44 @@
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| awsec2packageupgrade | This is the command Id of the command initiated by AWS SSM command | Unknown |
```
```suggestion
| awsec2packageupgrade | The command ID of the command initiated by AWS SSM command. | Unknown |
```

ShirleyDenkberg commented:
@johnnywilkes @Shellyber @ssokolovich Doc review completed.

capanw commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,195 @@
+    asm_rule_id: str,
+    region: str,
+    assume_role: str,
+    ROLE_SESSION_NAME: str,
```
Updated!!

capanw commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,195 @@
+                {"roleArn": assume_role, "roleSessionName": ROLE_SESSION_NAME}
+            )
+        output = demisto.executeCommand("aws-ssm-command-run", cmd_args)
+        output_run_command_dict["run_command_output"] = (
```
Added this suggestion. The ID will now be in run_command_id.

capanw commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,541 @@
+    quietmode: 0
+    scriptarguments:
+      command_id:
+        simple: ${awsec2packageupgrade.runcommandoutput}
```
Updated!!

BigEasyJ commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
Suggestion: 
- Make new playbook input for AWS project / account ID.
- Pass in the AWS project / account ID to this playbook as an input, to be used in the `AWSEC2PackageUpgradeScript`.

BigEasyJ commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
Suggestion: 
- Add input for AWS project ID to be supplied by the playbook input.
- Add a functionality in existing function or new function to build the ARN for the role.

BigEasyJ commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,198 @@
+    instance_id: str,
+    asm_rule_id: str,
+    region: str,
+    assume_role: str
```
```suggestion
    assume_role_arn: str
```

BigEasyJ commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,198 @@
+        instance_id (str): The ID of the instance where the package will be upgraded.
+        asm_rule_id (str): The ID of the ASM rule that specifies the package to be upgraded.
+        region (str): The AWS region where the instance is located.
+        assume_role (str): The AWS IAM role that will be assumed.
```
```suggestion
        assume_role_arn (str): The AWS IAM role that will be assumed.
```

BigEasyJ commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,198 @@
+
+    # Get the instance information
+    cmd_args = {"instance_id": instance_id, "type_name": "Instance Information"}
+    if assume_role:
```
```suggestion
    if assume_role_arn:
```

BigEasyJ commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,198 @@
+    # Get the instance information
+    cmd_args = {"instance_id": instance_id, "type_name": "Instance Information"}
+    if assume_role:
+        cmd_args.update({"roleArn": assume_role, "roleSessionName": ROLE_SESSION_NAME})
```
```suggestion
        cmd_args.update({"roleArn": assume_role_arn, "roleSessionName": ROLE_SESSION_NAME})
```

johnnywilkes reviewed (approved):
Thank you for making the changes I recommended.

capanw commented:
@BigEasyJ, Updated with the provided suggestions

capanw commented:
@ShirleyDenkberg @ssokolovich @Shellyber This is ready for review and to be merged. Thanks

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,549 @@
+description: This playbook upgrades supported package on an AWS EC2 instance.
```
```suggestion
description: This playbook upgrades supported packages on an AWS EC2 instance.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade.yml:
```diff
@@ -0,0 +1,549 @@
+      description: Determines if the AWS - Systems Manager integration instance is configured and input values are defined.
+      id: 75306cbe-5048-4052-8a48-ac85290e53e4
+      iscommand: false
+      name: Is AWS - Systems Manager enabled and are Input values defined?
```
```suggestion
      name: Is AWS - Systems Manager enabled and are input values defined?
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade_README.md:
```diff
@@ -0,0 +1,48 @@
+
+### Sub-playbooks
+
+* GenericPolling
```
```suggestion
GenericPolling
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade_README.md:
```diff
@@ -0,0 +1,48 @@
+
+### Integrations
+
+* AWS - System Manager
```
```suggestion
AWS - System Manager
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_EC2_Package_Upgrade_README.md:
```diff
@@ -0,0 +1,48 @@
+
+### Commands
+
+* aws-ssm-command-list
```
```suggestion
aws-ssm-command-list
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/README.md:
```diff
@@ -1,18 +1,19 @@
 The pack contains AWS playbooks that conduct enrichment and/or remediation and can use multiple other AWS content packs:
-- Enrichment: Give an IP address, see if there is a EC2 instance associated and if so pull information on the security group associated.
-- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved.
+- Enrichment: Give an IP address, see if there is an associated EC2 instance and if so pull information on the associated security group. If an SSM agent is installed on the instance, pull information from the SSM agent.
```
```suggestion
- Enrichment: Given an IP address, see if there is an associated EC2 instance and if so pull information on the associated security group. If an SSM agent is installed on the instance, pull information from the SSM agent.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/README.md:
```diff
@@ -1,18 +1,19 @@
-- Enrichment: Give an IP address, see if there is a EC2 instance associated and if so pull information on the security group associated.
-- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved.
+- Enrichment: Give an IP address, see if there is an associated EC2 instance and if so pull information on the associated security group. If an SSM agent is installed on the instance, pull information from the SSM agent.
+- Remediation: Give the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved. If there is a matching ASM rule ID that is supported, upgrade the associated package to the latest available version.
```
```suggestion
- Remediation: Given the information collected from enrichment, replace the security group with a "quarantine" security group until vulnerabilities are resolved. If there is a matching ASM rule ID that is supported, upgrade the associated package to the latest available version.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_18.md:
```diff
@@ -0,0 +1,16 @@
+
+##### AWS - Enrichment
+
+Updated the playbook to enrich using AWS SSM.
```
```suggestion
Updated the playbook to enrich EC2 and IAM information using AWS SSM.
```

ShirleyDenkberg commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.yml:
```diff
@@ -0,0 +1,54 @@
+name: AWSEC2PackageUpgrade
+outputs:
+- contextPath: awsec2packageupgrade
+  description: The command ID of the command initiated by AWS SSM command.
```
```suggestion
  description: The command ID of the command initiated by the AWS SSM command.
```

ShirleyDenkberg commented:
@johnnywilkes @Shellyber @ssokolovich Doc review completed.

Shellyber commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,204 @@
+        was run successfully, and 'run_command_output' containing the output of the command or error message.
+    """
+
+    output_run_command_dict = {"run_command_flag": True, "run_command_output": ""}
```
I'm not sure why you need this. 
Why not return the HR for each case?

Also, why not assign the flag to False, and then only when it succeeds to True? 
IT seems that most of your cases it is False.

capanw replied:
Hello @Shellyber, we want to return a dict always with a flag included. We plan on using this output in a playbook logic within our pack.

There was no particular reason for setting the flag True.

Shellyber commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.py:
```diff
@@ -0,0 +1,204 @@
+            + " "
+            + instance_info_dict.get("PlatformName", "")
+        )
+        # if os not in asm_rule_package_dict.get(asm_rule_id).keys():
```
You can remove the comment here

capanw replied:
Thanks for catching this. Removed this.

Shellyber reviewed (commented):
Nice one

Shellyber commented on Packs/AWS-Enrichment-Remediation/Scripts/AWSEC2PackageUpgrade/AWSEC2PackageUpgrade.yml:
You are missing the `fromversion` field here (should be 6.10.0)

capanw replied:
Updated!!

capanw commented on Packs/AWS-Enrichment-Remediation/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "description": "Playbooks using multiple AWS content packs for enrichment and remediation purposes",
     "support": "xsoar",
-    "currentVersion": "1.1.17",
+    "currentVersion": "1.1.18",
```
Updated

capanw commented:
@Shellyber, Updated with the suggestions provided.

ssokolovich reviewed (commented):
Hey @capanw - thanks for this contribution. 
Overall this looks good - A couple of notes/thoughts from my side:
1. Why do we need a dedicated script that calls for an EC2 package upgrade if we can leverage the existing integration to do so ( which is located in the `AWS - EC2` pack)?
2. Additionally, I suggest moving the `AWS - EC2 Package upgrade` playbook and its README+image to the `AWS - EC2` pack, which fits this PB better. (this will require also reverting the AWS - Enrichment pack README file as well)
3. Once we agreed on the above, I would like us to have a quick discussion over a meeting about the `AWS - EC2 Package upgrade`. I see that you are lavreging the SystemManager there, but can we do the same with the EC2 integration instead?
Moreover, I want to avoid using the GenericPolling playbook and instead maybe add that you a future command/to your upgrade script (instead of checking that action status from the PB).

Let me know your thoughts.
Cheers!

capanw commented:
> Hey @capanw - thanks for this contribution. Overall this looks good - A couple of notes/thoughts from my side:
> 
> 1. Why do we need a dedicated script that calls for an EC2 package upgrade if we can leverage the existing integration to do so ( which is located in the `AWS - EC2` pack)?
> 2. Additionally, I suggest moving the `AWS - EC2 Package upgrade` playbook and its README+image to the `AWS - EC2` pack, which fits this PB better. (this will require also reverting the AWS - Enrichment pack README file as well)
> 3. Once we agreed on the above, I would like us to have a quick discussion over a meeting about the `AWS - EC2 Package upgrade`. I see that you are lavreging the SystemManager there, but can we do the same with the EC2 integration instead?
>    Moreover, I want to avoid using the GenericPolling playbook and instead maybe add that you a future command/to your upgrade script (instead of checking that action status from the PB).
> 
> Let me know your thoughts. Cheers!

Hello @ssokolovich, thank you so much for reviewing and providing suggestions

- We wanted to have a dedicated script to simplify some of the steps in our main playbook otherwise, it would result in lot of branches in the playbook. We also plan on adding more functionality to this playbook (like supporting more use cases for remediation). The script doesn't use any commands from AWS - EC2 pack integration commands, the info that we need can only come from AWS - SSM integration.
- We use this script for enrichment and remediation and hence, we placed this in the AWS-Enrichment-Remediation pack.
- We used GenericPolling playbook, as it was super easy and didn't had to do the heavy lifting.

I can setup a meeting to go over these changes. Let me know if that works. Thanks

capanw commented:
Hello @ssokolovich @Shellyber, as discussed during our call today, I made changes to the following:

- Updated the Script?playbook names and removed EC2, to avoid any confusion.
- Moved Script/Playbook to the AWS Systems manager pack.

Thank you for your time and providing feedback!!!

Shellyber commented on Packs/AWS_SystemManager/Playbooks/AWS_-_Package_Upgrade_README.md:
```diff
@@ -0,0 +1,48 @@
+
+### Scripts
+
+* AWSEC2PackageUpgrade
```
```suggestion
* AWSPackageUpgrade
```

capanw replied:
Updated.

Shellyber commented on Packs/AWS-Enrichment-Remediation/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "description": "Playbooks using multiple AWS content packs for enrichment and remediation purposes",
     "support": "xsoar",
-    "currentVersion": "1.1.17",
+    "currentVersion": "1.2.0",
```
```suggestion
    "currentVersion": "1.1.18",
```
Please don't forget to change the release notes file name as well

capanw replied:
Updated the versions.

Shellyber commented on Packs/AWS_SystemManager/ReleaseNotes/1_0_2.md:
```diff
@@ -0,0 +1,17 @@
+
+New: This is an AWS script that upgrades a package on the AWS EC2 instance using AWS Systems manager.
+
+#### Pack Metadata
```
You can remove this part

capanw replied:
Done

Shellyber reviewed (commented):
Looking good for my part. 
last small notes. 
Also, pre-commit fails on `RM104`.
Best practice is to add the README as suggested, if not you can put the code in the ignore file.

ssokolovich commented on Packs/AWS_SystemManager/doc_files/AWSPackageUpgrade.png:
not sure if you saw this but the image has a timestamp and the left corner - maybe we should remove that?

capanw replied:
Removed the timestamp and pushed new image

Shellyber commented on Packs/AWS_SystemManager/Playbooks/AWS_-_Package_Upgrade.yml:
```diff
@@ -0,0 +1,549 @@
+      id: 5f029ce5-bc0d-4555-800c-47be2c8ea935
+      iscommand: false
+      name: 'AWS Package Upgrade '
+      script: AWSEC2PackageUpgrade
```
Need to update the script here as well

capanw replied:
Updated!!

capanw commented:
Hello @Shellyber, the precommit is failing due to AWS Systems manager README. I left it empty and have not made any changes now and it still throwing validation error. Can this be bypassed?

Shellyber commented:
@capanw Just add the error code `RM104` to the pack_ignore file

capanw commented:
Hello @Shellyber, added to pack ignore and I think this looks good to be merged!! Let me know if there is anything else. Thank you
